analyzers/src/SonarAnalyzer.Common/Rules/RouteTemplateShouldNotStartWithSlashBase.cs

```diff
@@ -0,0 +1,107 @@
+            {
+                return;
+            }
+
+            compilationStartContext.RegisterSymbolStartAction(symbolStartContext =>
+            {
+                var symbol = (INamedTypeSymbol)symbolStartContext.Symbol;
+                if (symbol.IsControllerType())
+                {
+                    var controllerActionInfo = new ConcurrentStack<ControllerActionInfo>();
+                    symbolStartContext.RegisterSyntaxNodeAction(nodeContext =>
+                    {
+                        if (nodeContext.SemanticModel.GetDeclaredSymbol(nodeContext.Node) is IMethodSymbol methodSymbol && methodSymbol.IsControllerMethod())
+                        {
+                            controllerActionInfo.Push(new ControllerActionInfo(methodSymbol, RouteAttributeTemplateArguments(methodSymbol.GetAttributes())));
+                        }
+                    }, MethodSyntaxKind);
+
+                    ReportIssues(symbolStartContext, symbol, controllerActionInfo);
+                }
+            }, SymbolKind.NamedType);
+        });
+
+    protected bool IsGeneratedCode(SyntaxReference syntaxReference) =>
+        syntaxReference.GetSyntax().SyntaxTree.IsGenerated(GeneratedCodeRecognizer, null);
+
+    private void ReportIssues(SonarSymbolStartAnalysisContext context, INamedTypeSymbol controllerSymbol, ConcurrentStack<ControllerActionInfo> actions) =>
+        context.RegisterSymbolEndAction(context =>
+        {
+            if (!actions.Any()
+                || actions.Any(x => !x.RouteParameters.Any() || x.RouteParameters.Keys.Any(x => !x.StartsWith("/"))))
+            {
+                return;
+            }
+            var issueMessage = controllerSymbol.GetAttributes().Any(x => x.AttributeClass.IsAny(KnownType.RouteAttributes))
+                ? MessageOnlyActions
+                : MessageActionsAndController;
+
+            var attributeLocations = actions.SelectMany(x => x.RouteParameters).Select(x => x.Value);
+            if (ControllerLocation(controllerSymbol) is { } controllerLocation)
+            {
+                context.ReportIssue(Language.GeneratedCodeRecognizer, Diagnostic.Create(Rule, controllerLocation, attributeLocations, issueMessage));
+            }
+        });
+
+    private static Dictionary<string, Location> RouteAttributeTemplateArguments(ImmutableArray<AttributeData> attributes)
+    {
+        var templates = new Dictionary<string, Location>();
+        var routeAttributes = attributes.Where(x => x.AttributeClass.IsAny(KnownType.RouteAttributes));
+        foreach (var attribute in routeAttributes)
+        {
+            if (attribute.TryGetAttributeValue<string>("template", out var templateParameter))
+            {
+                templates.Add(templateParameter, attribute.ApplicationSyntaxReference.GetSyntax().GetLocation());
+            }
+        }
+        return templates;
+    }
+
+    private static bool ShouldRegisterAction(Compilation compilation) =>
```

Comment:
The same check is done in S6931 (`BackslashShouldBeAvoidedInAspNetRoutesBase`), except that that check also checks for ASP.NET Core controller, which this rule doesn't support yet. So:
- if moving the implementation of `BackslashShouldBeAvoidedInAspNetRoutesBase.ShouldRegisterAction` (that checks for both ASP.NET 4.x and ASP.NET Core controller types) into a `SonarAnalyzer.Helpers.AspNetHelper` and using that helper in this rule doesn't break this rule, we should probably do that
- if it somehow does break this rule (because the rule would register also in ASP.NET Core), this comment can be ignored   

I think that the factorization should not break this rule, since we would just be registering for SymbolStart more often. To be checked.